README.md

```diff
@@ -3,10 +3,126 @@
 This Kubewarden Policy is a replacement for the Kubernetes Pod Security 
 Policy that controls containers [user and groups](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#users-and-groups).
 
+This policy is used to control the user and groups user in containers.
+
+## Settings
+
+
+The policy has three settings:
+
+* `run_as_user`: Controls which user ID the containers are run with. It is a JSON object with two attributes: `rule` and `ranges`. The `rule` allowed values are:
+* `run_as_group`:  Controls which primary group ID the containers are run with.
+* `supplemental_groups`: Controls which group IDs containers add.
+
+All the three settings are JSON objects composed by two attributes: `rule` and `ranges`. The `rule` attribute define
+the strategy used by the policy to enforce user and groups used in containers. The available strategies are:
+
+* `run_as_user`: 
+	* `MustRunAs` - Requires at least one range to be specified. Uses the minimum value of the first range as the default. Validates against all ranges.
+	* `MustRunAsNonRoot` - Requires that the pod be submitted with a non-zero `runAsUser` or have the `USER` directive defined (using a numeric UID) in the image. Pods which have specified neither `runAsNonRoot` nor `runAsUser` settings will be mutated to set `runAsNonRoot=true`, thus requiring a defined non-zero numeric `USER` directive in the container. No default provided. 
+	* `RunAsAny` - No default provided. Allows any `runAsUser` to be specified.
+* `run_as_group`: 
+	* `MustRunAs` - Requires at least one range to be specified. Uses the minimum value of the first range as the default. Validates against all ranges.
+	* `MayRunAs` - Does not require that `RunAsGroup` be specified. However, when `RunAsGroup` is specified, they have to fall in the defined range.
+	* `RunAsAny` - No default provided. Allows any `runAsGroup` to be specified.
+* `supplemental_groups`: 
+	* `MustRunAs` - Requires at least one range to be specified. Uses the minimum value of the first range as the default. Validates against all ranges.
+	* `MayRunAs` - Requires at least one range to be specified. Allows `supplementalGroups` to be left unset without providing a default. Validates against all ranges if `supplementalGroups` is set.
+	* `RunAsAny` - No default provided. Allows any `supplementalGroups` to be specified
+
+The `ranges` is a list of JSON objects with two attributes: `min` and `max`. Each range object define the user/group ID range used by the rule. 
+
+### Examples
+
+To enforce that user and groups must be set and it should be in the defined ranges:
+
+```json
+{
+        "run_as_user": {
+                "rule": "MustRunAs",
+                "ranges": [
+                        {"min": 1000, "max": 2000},
+                        {"min": 2001, "max": 3000}
```

Comment:
In the example I would add a gap, so it's clear the usage. For example:

```
                        {"min": 1000, "max": 1999},
                        {"min": 3000, "max": 3999}
```

In this example, 2000-2999 is out of any allowed ranges, and would be rejected.